### PR TITLE
Add files_sharing capabilities

### LINKED ISSUE #1754: Add capability send_password_by_mail
https://github.com/nextcloud/server/pull/29332/files#diff-2f474cd876a5cee804e810c6b97cba8d45cc43afdce5ba7d17e037de4a150f12R68

adds a new capability send_password_by_mail for >= NC23:

show this capability only if it exists :)

### PR DESCRIPTION
**Pull request by thisIsTheFoxe**
Add json paths for 
- `send_password_by_mail` capability bool value
  - Close #1754 
- `default_permissions` int value
  - #1701 
  - https://github.com/nextcloud/files-clients/issues/4

Usage:
`
NCManageDatabase.shared.getCapabilitiesServerBool(account: <#User account#>, elements: NCElementsJSON.shared.capabilitiesFileSharingSendPasswordMail, exists: false)
`

### CODE REVIEW

**Comment by marinofaggiana** [comment]
https://github.com/nextcloud/ios/blob/7188c577788a68aa2ec97191ee17ae849cd2ba40/iOSClient/Data/NCElementsJSON.swift#L37

Without the ":" ?

File: iOSClient/Data/NCElementsJSON.swift [hunk 1]
```diff
@@ -34,6 +34,9 @@ import UIKit
     
     @objc public let capabilitiesFileSharingApiEnabled:         Array = ["ocs","data","capabilities","files_sharing","api_enabled"]
     @objc public let capabilitiesFileSharingPubPasswdEnforced:  Array = ["ocs","data","capabilities","files_sharing","public","password","enforced"]
+    @objc public let capabilitiesFileSharingDefaultPermissions  Array = ["ocs","data","capabilities","files_sharing","default_permissions"]
+    // NC >= 23
```
Comment (marinofaggiana) [inline iOSClient/Data/NCElementsJSON.swift:38]: Line 37 - ":"

Comment (thisIsTheFoxe) [reply iOSClient/Data/NCElementsJSON.swift:38]: Good catch.. Made this in the GH editor without building 😅

Comment (marinofaggiana) [reply iOSClient/Data/NCElementsJSON.swift:38]: But where you use it ?

**Review by marinofaggiana — APPROVED** [review]

**Pull request merged**